cmd/jujud/agent/machine.go

```diff
@@ -225,14 +225,24 @@ func (a *machineAgentCmd) Init(args []string) error {
 		return errors.Annotate(err, "cannot read agent configuration")
 	}
 
+	config := a.currentConfig.CurrentConfig()
 	// the context's stderr is set as the loggo writer in github.com/juju/cmd/logging.go
 	a.ctx.Stderr = &lumberjack.Logger{
-		Filename:   agent.LogFilename(a.currentConfig.CurrentConfig()),
+		Filename:   agent.LogFilename(config),
 		MaxSize:    300, // megabytes
 		MaxBackups: 2,
 		Compress:   true,
 	}
 
+	if loggingOverride := config.Value(agent.LoggingOverride); loggingOverride != "" {
```

Comment:
The rationale behind not adding the logging override to the Validate function is that it is entirely a debugging option, and any value that is set will be checked when the agent starts anyway, and if it doesn't work, an error will be omitted there. Adding another logged error seems a bit pointless.